src/geoserver/style.py

```diff
@@ -131,8 +131,19 @@ def sld_body(self):
 
     @property
     def body(self):
-        # [:-2] remove version tag from type. GeoServer does not accept it
-        resp = self.catalog.http_request(self._build_href('.{}'.format(self.style_format[:-2])))
+        href_ext = ''
+        _headers = {}
+        if self.style_format and Style.content_types.get(self.style_format):
```

Comment:
@giohappy within this block the `href_ext` won't be valorized... is this wanted?

Reply:
yes @afabiani. The problem was that setting the .sld extension Geoserver treats the style as sld10 (whatever the real format is). The extension must be removed and the headers must be set to force Geoserver to return the original format.